ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -321,7 +323,7 @@ private Callback<Void> routerTtlUpdateCallback(BlobInfo blobInfo, String blobId)
           deleteDatasetVersionOutOfRetentionCount();
         }
         securityService.processResponse(restRequest, restResponseChannel, blobInfo, securityProcessResponseCallback());
-      }, uri, LOGGER, finalCallback);
+      }, uri, LOGGER, deleteDatasetCallback);
     }
```

Comment:
For the last step, securityProcessResponseCallback, shall we change the failure callback to deleteDatasetCallback as well?

I feel for any buildCallback after the PutBlob step, we should replace the failure callback to deleteDatasetCallback.

```   
private Callback<Void> securityProcessResponseCallback() {
      return buildCallback(frontendMetrics.putBlobSecurityProcessResponseMetrics,
          securityCheckResult -> finalCallback.onCompletion(null, null), restRequest.getUri(), LOGGER, finalCallback);
```

Reply:
This step is to sumbit the response. the named blob and regular blob already in our service. So I feel like no need to rollback if only the sumbit response failed. Instead, client will take the best effort to manually delete this version.